### PR TITLE
Add Mask R-CNN

### PR DESCRIPTION
**Pull request by ijkguo**
Main changes:
- Add additional outputs to Faster R-CNN models for Mask R-CNN training and inference.
- Add Mask R-CNN.

### CODE REVIEW

File: gluoncv/data/transforms/mask.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
+"""Mask transformation functions."""
+import copy
+import cv2
+import numpy as np
+from ..mscoco.utils import try_import_pycocotools
+# pylint: disable=wrong-import-position,wrong-import-order
+try_import_pycocotools()
```
Comment (zhreshold) [inline gluoncv/data/transforms/mask.py]: import only when required

Comment (ijkguo) [reply gluoncv/data/transforms/mask.py]: Moved inside to_mask function

**Comment by mli** [comment]
Job PR-252-10 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-252/10/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-252-11 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-252/11/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-252-12 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-252/12/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-252-15 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-252/15/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-252-16 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-252/16/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-252-17 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-252/17/index.html 
Code coverage of this PR:  vs. Master:

**Pull request merged**